src/main/java/edu/harvard/iq/dataverse/api/Access.java

```diff
@@ -423,7 +423,10 @@ public DownloadInstance datafile(@PathParam("fileId") String fileId, @QueryParam
         /* 
          * Provide some browser-friendly headers: (?)
          */
-        return downloadInstance;
+        if (headers.getRequestHeaders().containsKey("Range")) {
+            return Response.status(Response.Status.PARTIAL_CONTENT).entity(downloadInstance).build();
+        }
+        return Response.ok(downloadInstance).build();
```

Comment:
Could you please clarify what this change does? 
Specifically, changing the return value from DownloadInstance to Response - are you positive that it's still going to stream the output? (As opposed to trying to read everything from the DownloadInstance into memory at once, and *then* build the Response?)

Reply:
Let's start with the requirement:
If the request contains a "Range" header and the requested range of bytes is returned, a HTTP status of 206 (Partial Content) instead of 200 must be returned. See also
https://datatracker.ietf.org/doc/html/rfc2616#section-14.35.2


So, these lines change the status code of the HTTP response dependent on the existence of the Range header in the HTTP request. The only way to change it is by using the Response object. (Setting it directly in the HttpServletResponse would be overwritten by Jersey later).

However, there is no real difference between returning the entity (DownloadInstance) directly, or wrapped in a Response object. Except that Response allows more control over settings like the status code. The custom MessageBodyWriter for DownloadInstance is still called, and that is where the streaming happens.

Reply:
Thanks. For the record, this is *very* old code. Written for a very early version of Jersey. 
We have an open issue for refactoring that download subsystem. But it's going to be a project, with years worth of special case handling added there.